pkg/cloud/services/eks/roles.go

```diff
@@ -304,7 +304,7 @@ func (s *FargateService) reconcileFargateIAMRole() (requeue bool, err error) {
 			s.scope.Info("no EKS fargate role specified, using role based on fargate profile name")
 			roleName, err = eks.GenerateEKSName(
 				"fargate",
-				fmt.Sprintf("%s-%s", s.scope.KubernetesClusterName(), s.scope.FargateProfile.Spec.ProfileName),
```

Comment:
Not sure if we should use `s.scope.FargateProfile.Spec.ClusterName` instead - unsure of the conventions at play.